internal.nix

```diff
@@ -33,6 +33,11 @@ rec {
   makeValidDrvName = str:
     lib.stringAsChars (c: if isValidDrvNameChar c then c else "?") str;
 
+  # Description: Checks if a string looks like a valid git revision
+  # Type: String -> Boolean
+  isGitRev = str:
```

Comment:
Do we have a good case that would *not* match this condition? I'd love to add a test that ensures we are never treating that as a git rev. If we can.

Reply:
I haven't seen anything in a lockfile that specifies a github dependency in the `from` field but does not contain a commit rev in the `version` field. But I also haven't looked at many lockfiles. So I don't know if there are good cases, that would not match.

Reply:
@andir is there any other reason for not merging this? I see that tests were added but it's idle almost a half year. I just encountered the ENOTCACHED bug and was wondering if this could fix it.